pkg/github/github.go

```diff
@@ -153,17 +153,16 @@ func (c *Client) uploadReleaseAsset(ctx context.Context, releaseID int64, filena
 		return errors.Wrap(err, "failed to get abs path")
 	}
 
-	f, err := os.Open(filename)
-	if err != nil {
-		return errors.Wrap(err, "failed to open file")
-	}
-
 	opts := &github.UploadOptions{
 		// Use base name by default
 		Name: filepath.Base(filename),
 	}
 
 	if err := retry.Retry(3, 3*time.Second, func() error {
+		f, err := os.Open(filename)
+		if err != nil {
+			return errors.Wrap(err, "failed to open file")
+		}
```

Comment:
It makes sense to move that here. Not sure if that alone fixes the issue. However, looking at this, I figured that the file is never closed, which could actually cause this.

```suggestion
		f, err := os.Open(filename)
		if err != nil {
			return errors.Wrap(err, "failed to open file")
		}
		defer f.Close()
```